tripper/triplestore.py

```diff
@@ -673,6 +680,153 @@ def prefix_iri(self, iri: str, require_prefixed: bool = False):
                 raise NamespaceError(f"No prefix defined for IRI: {iri}")
         return iri
 
+    # Types of restrictions defined in OWL
+    _restriction_types = {
+        "some": (OWL.someValueFrom, None),
+        "only": (OWL.allValueFrom, None),
+        "exactly": (OWL.onClass, OWL.qualifiedCardinality),
+        "min": (OWL.onClass, OWL.minQualifiedCardinality),
+        "max": (OWL.onClass, OWL.maxQualifiedCardinality),
+        "value": (OWL.hasValue, None),
+    }
+
+    def add_restriction(  # pylint: disable=redefined-builtin
+        self,
+        cls: str,
+        property: str,
+        target: "Union[str, Literal]",
+        type: "TypeLiteral['some', 'only', 'exactly', 'min', 'max', 'value']",
+        cardinality: "Optional[int]" = None,
+        hashlength: int = 16,
```

Comment:
Should not this also be typeset to optional?

Reply:
I don't think so. `hashlength` must always be an integer. The "Optional" type annotation indicates that you can provide None as a default value. That is not allowed for `hashlength`.